lib/function/index.js

```diff
@@ -0,0 +1,247 @@
+    if (n === 1) return parameters.stops[0][1];
+    if (input === undefined || input === null) return parameters.stops[n - 1][1];
+    if (input <= parameters.stops[0][0]) return parameters.stops[0][1];
+    if (input >= parameters.stops[n - 1][0]) return parameters.stops[n - 1][1];
+
+    var index = binarySearchForIndex(parameters.stops, input);
+
+    return interpolate(
+            input,
+            base,
+            parameters.stops[index][0],
+            parameters.stops[index + 1][0],
+            parameters.stops[index][1],
+            parameters.stops[index + 1][1]
+    );
+}
+
+function evaluateIdentityFunction(parameters, input) {
+    return input;
+}
+
+function binarySearchForIndex(stops, input) {
+  var n = stops.length;
+  var lowerIndex = 0;
+  var upperIndex = n - 1;
+  var currentIndex = 0;
+  var currentValue;
+
+  while (lowerIndex <= upperIndex) {
+    currentIndex = Math.floor((lowerIndex + upperIndex) / 2);
+    currentValue = stops[currentIndex][0];
+    if (currentValue === input) {
+      currentIndex += 1;
+      break;
+    } else if (currentValue < input) {
+      lowerIndex = currentIndex + 1;
+    } else if (currentValue > input) {
+      upperIndex = currentIndex - 1;
+    }
+  }
+
+  return Math.max(currentIndex - 1, 0);
+}
+
+function interpolate(input, base, inputLower, inputUpper, outputLower, outputUpper) {
+    if (typeof outputLower === 'function') {
+        return function() {
+            var evaluatedLower = outputLower.apply(undefined, arguments);
+            var evaluatedUpper = outputUpper.apply(undefined, arguments);
+            return interpolate(input, base, inputLower, inputUpper, evaluatedLower, evaluatedUpper);
+        };
+    } else if (outputLower.length) {
+        return interpolateArray(input, base, inputLower, inputUpper, outputLower, outputUpper);
+    } else {
+        return interpolateNumber(input, base, inputLower, inputUpper, outputLower, outputUpper);
+    }
+}
+
+function interpolateNumber(input, base, inputLower, inputUpper, outputLower, outputUpper) {
+    var difference = inputUpper - inputLower;
+    var progress = input - inputLower;
+
+    var ratio;
+    if (base === 1) {
+        ratio = progress / difference;
+    } else {
+        ratio = (Math.pow(base, progress) - 1) / (Math.pow(base, difference) - 1);
+    }
+
+    return (outputLower * (1 - ratio)) + (outputUpper * ratio);
+}
+
+function interpolateArray(input, base, inputLower, inputUpper, outputLower, outputUpper) {
+    var output = [];
+    for (var i = 0; i < outputLower.length; i++) {
+        output[i] = interpolateNumber(input, base, inputLower, inputUpper, outputLower[i], outputUpper[i]);
+    }
+    return output;
+}
+
+function isFunctionDefinition(value) {
+    return typeof value === 'object' && (value.stops || value.type === 'identity');
+}
+
+
+module.exports.isFunctionDefinition = isFunctionDefinition;
+
+module.exports.interpolated = function(parameters) {
```

Comment:
The addition of property functions doesn't eliminate the need to know whether a function is `interpolated` or `piecewise-constant`.

It would be possible to refactor to make this an argument on one function instead of two functions. I wasn't planning to do this refactor until a compelling reason arose.